### PR TITLE
First attempt at providing solutions directory and stubbed code for Kafka Streams

### PR DESCRIPTION
**Pull request by bbejeck**
This PR is a proof of concept for stubbing out the Kafka Streams code and moving the fully completed exercises to a `solutions` directory/package

### CODE REVIEW

File: kafka-streams/src/main/java/io/confluent/developer/aggregate/StreamsAggregate.java [hunk 1]
```diff
@@ -31,16 +31,24 @@ public static void main(String[] args) throws IOException {
...
+              // Now take the electronicStream object, group by key and perform an aggregation
+              // Don't forget to convert the KTable returned by the aggregate call back to a KStream using the toStream method
+              electronicStream.groupByKey().aggregate(null, null);
+
+              // To view the results of the aggregation consider
+              // right after the toStream() method .peek((key, value) -> System.out.println("Outgoing record - key " +key +" value " + value))
+
+              // Finally write the results to an output topic
+              // using a KStream.to method with a Produced config object
+              // and the appropriate Serdes for the key Serdes.String()  and the value Serdes.Long()
```
Comment (bbejeck) [inline kafka-streams/src/main/java/io/confluent/developer/aggregate/StreamsAggregate.java:51]: This is how the exercises will look now, just a few stubs and comments on what's needed to add to make it functional.

File: kafka-streams/README.md [hunk 2]
```diff
@@ -4,27 +4,33 @@ This is the code repo containing the full solutions to the exercises
 for the Kafka Streams course.  To get started you'll need to sign up
 for a CCloud account.  Follow the steps below to get everything ready to go.
```
Comment (rspurgeon) [inline kafka-streams/README.md:5]:
```suggestion
for a Confluent Cloud account.  Follow the steps below to get everything ready to go.
```

File: kafka-streams/README.md [hunk 3]
```diff
@@ -4,27 +4,33 @@ This is the code repo containing the full solutions to the exercises
...
+It's recommended that you do these exercises while following along with the [Kafka Streams 101 course](https://developer.confluent.io/learn-kafka/kafka-streams/get-started/).
+The link for each module of the course related to the exercise description will be provided.
+
 ##  Confluent Cloud
  
 1. Go to [Confluent Cloud](https://www.confluent.io/confluent-cloud/tryfree?utm_source=learnkafka). 
    If you don’t already have a Confluent Cloud account you can create one here. Use promotion code <TBC> for money off your cloud bill.
-   
-2. Create a new cluster in Confluent Cloud.  For the purposes of all the exercise modules you can use the `Basic` type.
+2. Once logged into your cloud account click on the `Add cloud environment button` on the right-hand side of the screen.  Name the environment something like "kafka_streams_course_env".  You can also choose to use the `default` environment already present.  Using a new environment gives you the ability to keep resources separate on your Confluent account.
```
Comment (rspurgeon) [inline kafka-streams/README.md:14]:
```suggestion
2. Once logged into your cloud account click on the `Add cloud environment button` on the right-hand side of the screen.  Name the environment something like `kafka_streams_course_env`.  You can also choose to use the `default` environment already present.  Using a new environment gives you the ability to keep resources separate on your Confluent account.
```

File: kafka-streams/README.md [hunk 4]
```diff
@@ -4,27 +4,33 @@ This is the code repo containing the full solutions to the exercises
...
+   ![Create an environment or use the default one](images/confluent_environment.png)
+3. Click on the environment you just created or the `default` environment tile. Create a new cluster in Confluent by clicking on`Create cluster on my own`.  For the purposes of all the exercise modules you can use the `Basic` type.
```
Comment (rspurgeon) [inline kafka-streams/README.md:16]:
```suggestion
3. Click on the environment you just created or the `default` environment tile. Create a new cluster by clicking on `Create cluster on my own`.  For the purposes of all the exercise modules you can use the `Basic` type. 
```

File: kafka-streams/README.md [hunk 5]
```diff
@@ -4,27 +4,33 @@ This is the code repo containing the full solutions to the exercises
...
-   4. Then click on the `Copy` link to the right immediately above the properties
-    5. Create a file named `ccloud.properties` in the `src/main/resources` directory. Then paste the configurations into the `ccloud.properties` file.  Note that
-    this file is ignored and should never get checked in.
+
+   1. Click on `Create Kafka cluster API key`.
+   2. Copy your key and secret, name the file, then click `Download and continue`. (Your credentials will populate into the configurations boilerplate.)
+   3. Click on `Create Schema Registry API key`.
+   4. Copy your key and secret, name the file, then click `Download and continue`. (Your credentials will populate into the configurations boilerplate.)
+   5. Make sure `Show API keys` is selected, then `Copy` the configurations in the window.
+   6. Create a file named `ccloud.properties` in the `src/main/resources directory` of the cloned repo. Then paste the configurations into a file named `ccloud.properties`. Note that this file is ignored and should never get checked into GitHub.
```
Comment (rspurgeon) [inline kafka-streams/README.md:33]:
```suggestion
   6. Create a file named `ccloud.properties` in the `src/main/resources` directory of the cloned repo. Then paste the configurations into a file named `ccloud.properties`. Note that this file is ignored and should never get checked into GitHub.
```

**Comment by rspurgeon** [comment]
@bbejeck I made some comments on the copy in the README.  Otherwise, the code is LGTM.  If you accept those changes to the README I can push through the PR.

**Comment by bbejeck** [comment]
Thanks for the review @rspurgeon - I've accepted all the suggestions

**Pull request merged**